ui/ducks/send/send.js

```diff
@@ -0,0 +1,1442 @@
+        // changes to be pulled in.
+        state.status = SEND_STATUSES.INVALID;
+      })
+      .addCase(initializeSendState.fulfilled, (state, action) => {
+        // writes the computed initialized state values into the slice and then
+        // calculates slice validity using the caseReducers.
+        state.account.address = action.payload.address;
+        state.account.balance = action.payload.nativeBalance;
+        state.asset.balance = action.payload.assetBalance;
+        state.gas.gasLimit = action.payload.gasLimit;
+        state.gas.gasPrice = action.payload.gasPrice;
+        state.gas.gasTotal = action.payload.gasTotal;
+        if (state.stage !== SEND_STAGES.UNINITIALIZED) {
+          slice.caseReducers.validateRecipientUserInput(state, {
+            payload: {
+              chainId: action.payload.chainId,
+              tokens: action.payload.tokens,
+            },
+          });
+        }
+        state.stage =
+          state.stage === SEND_STAGES.UNINITIALIZED
+            ? SEND_STAGES.ADD_RECIPIENT
+            : state.stage;
+        slice.caseReducers.validateAmountField(state);
+        slice.caseReducers.validateGasField(state);
+        slice.caseReducers.validateSendState(state);
+      })
+      .addCase(computeEstimatedGasLimit.pending, (state) => {
+        // When we begin to fetch gasLimit the slice should be marked invalid
+        state.status = SEND_STATUSES.INVALID;
+      })
+      .addCase(computeEstimatedGasLimit.fulfilled, (state, action) => {
+        // When we receive a new gasLimit from the computeEstimatedGasLimit
+        // thunk we need to update our gasLimit in the slice. We call into the
+        // caseReducer updateGasLimit to tap into the appropriate follow up
+        // checks and gasTotal calculation.
+        if (action.payload?.gasLimit) {
+          slice.caseReducers.updateGasLimit(state, {
+            payload: action.payload.gasLimit,
+          });
+        }
+      })
+      .addCase(SET_BASIC_GAS_ESTIMATE_DATA, (state, action) => {
+        // When we receive a new gasPrice via the gas duck we need to update
+        // the gasPrice in our slice. We call into the caseReducer
+        // updateGasPrice to also tap into the appropriate follow up checks
+        // and gasTotal calculation.
+        slice.caseReducers.updateGasPrice(state, {
+          payload: getGasPriceInHexWei(action.value.average ?? '0x0'),
+        });
+      })
+      .addCase(BASIC_GAS_ESTIMATE_STATUS, (state, action) => {
+        // When we fetch gas prices we should temporarily set the form invalid
+        // Once the price updates we get that value in the
+        // SET_BASIC_GAS_ESTIMATE_DATA extraReducer above. Finally as long as
+        // the state is 'READY' we will revalidate the form.
+        switch (action.value) {
+          case BASIC_ESTIMATE_STATES.FAILED:
+            slice.caseReducers.useInlineGasFields(state);
```

Comment:
While it would be true in the context of this reducer, I'm taking advantage of the fact that status is set at the end of the fetchBasicGasEstimates method in gas.duck.js

https://github.com/MetaMask/metamask-extension/blob/f649bbb2831668cd40b61839c22a2e869eb14233/ui/ducks/gas/gas.duck.js#L113-L139

So, the swap back to basic gas fields happens in the reducer under this one, which is on the SET_ESTIMATE_SOURCE action which happens before the STATUS is set.